queries.sql

```diff
@@ -80,4 +80,24 @@ ROLLBACK;
   -- List all animals owned by Dean Winchester that haven't tried to escape.
   SELECT animals.name, owners.full_name FROM animals JOIN owners ON animals.owner_id = owners.id WHERE escape_attempts = 0 AND owners.id = 5;
   -- Who owns the most animals?
-  SELECT owners.full_name, COUNT(*) FROM owners LEFT JOIN animals ON owners.id = animals.owner_id GROUP BY owners.full_name ORDER BY count DESC;
\ No newline at end of file
+  SELECT owners.full_name, COUNT(*) FROM owners LEFT JOIN animals ON owners.id = animals.owner_id GROUP BY owners.full_name ORDER BY count DESC;
+
+  -- Write queries to answer the following:
+  -- Who was the last animal seen by William Tatcher?
+  SELECT animals.name, visits.date_of_visit FROM animals JOIN visits ON animals.id = visits.animal_id WHERE vet_id = 1 ORDER BY date_of_visit DESC;
+  -- How many different animals did Stephanie Mendez see?
+  SELECT COUNT(*) FROM animals JOIN visits ON animals.id = visits.animal_id WHERE vet_id = 3;
+  -- List all vets and their specialties, including vets with no specialties.
+  SELECT vets.name, species.name AS specialized_in FROM vets LEFT JOIN specializations ON specializations.vet_id = vets.id LEFT JOIN  species ON specializations.species_id = species.id;
+  -- List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
+  SELECT name FROM animals JOIN visits ON animals.id = visits.animal_id WHERE vet_id = 3 AND date_of_visit BETWEEN '2020/04/01' AND '2020/08/30';
+  -- What animal has the most visits to vets?
+  SELECT animals.name, COUNT(*) FROM animals JOIN visits ON animals.id = visits.animal_id GROUP BY animals.name ORDER BY count DESC LIMIT 1;
+  -- Who was Maisy Smith's first visit?
+  SELECT animals.name, visits.date_of_visit FROM animals JOIN visits ON animals.id = visits.animal_id WHERE vet_id = 2 GROUP BY animals.name, date_of_visit ORDER BY date_of_visit LIMIT 1;
+  -- Details for most recent visit: animal information, vet information, and date of visit.
+  SELECT animals.name, animals.date_of_birth, animals.escape_attempts, animals.neutered, animals.weight_kg, vets.name, vets.age, vets.date_of_graduation, visits.date_of_visit FROM animals JOIN visits ON animals.id = visits.animal_id JOIN vets ON vets.id = visits.vet_id ORDER BY date_of_visit DESC;
```

Comment:
- [ ] Please return a single animal who was most recently visited as mentioned in the requirements. Add `LIMIT ` after `DESC`. Currently, you are returning more than one animal.
![image](https://github.com/fatmahussein/vet_clinic/assets/45460192/d3628375-0136-47e8-b7e7-1fa13d318411)